### ISSUE TITLE
Factor levels get lost when plotting comparison arrows

### ISSUE THREAD

**Issue by czp-jlu** [rvlenth/emmeans#225]
Dear authors,

I noted that when plotting comparison arrows, the order of the factor levels gets lost.

Example:

```
library(emmeans)

dta <- data.frame(x.a = c(1.5, 2.5, 2, 7, 5, 5, 3, 5, 3),
                  x.b = factor(rep(c("a", "b", "c"), each = 3)))
dta$x.b <- factor(dta$x.b, levels = c("b", "c", "a"))

m1 <- lm(x.a ~ x.b, data = dta)
emm <- emmeans(m1, ~x.b)

plot(emm, CIs = F)
plot(emm, CIs = F, comparisons = T)
```

In the first plot, the order of the factor levels on the y axis corresponds to the order in the data frame. In the second plot with the comparison arrows, the factor levels are ordered alphanumerically. Is there a way to retain the order of the factor levels even with comparison arrows? I know this did work with emmeans 1.2.3.

Best regards
Carola

**Comment by rvlenth** [comment]
Hello,

I confirm this discrepancy. I recall making some changes a few versions ago (details are hazy, might have been the one in the NEWS for version 1.3.0) and that may be when this happened. I'll look at the code; I suspect that I construct a new factor based on combining potentially more than one factor, and that the levels goes to the default alphabetical ordering. I'll take a look and see if I can fix this. Thanks for the report.

Russ

**Comment by rvlenth** [comment]
Here are more clues:

                plot(emm, CI=F, comp=T, engine="lattice")  # ordering is not lost
                plot(emm, CI=F, comp=T, engine="ggplot")  # ordering is lost
                plot(emm, CI=T, comp=T, engine="ggplot")  # ordering is not lost

so it has to do with the way ggplot() orders grouping levels. With CI = T, it has an earlier chance to get it right...

And I wonder if your previous time it worked with ver. 1.2.3 was using the lattice engine.

Russ

**Comment by rvlenth** [comment]
Hi,

I've got this fixed; just plot invisible points at the start so the ordering comes out right regardless.

I'll be updating to CRAN soon, then it will be right.

Russ

**Comment by rvlenth** [comment]
Closing - fixed in version 1.5.1

**Comment by czp-jlu** [comment]
Thanks for the quick fix!
